### DISCUSSION TITLE
Youtube live stream quality issues?

### DISCUSSION THREAD

**Discussion by ne0rrmatrix** [Tyrrrz/YoutubeExplode#698] [Q&A] [answered]
I have managed to get a live stream using:
```
var youtube = new YoutubeClient();
        mediaElement.Source = await youtube.Videos.Streams.GetHttpLiveStreamUrlAsync("F2NreNEmMy4");
        mediaElement.Play();
```
This works for my dotnet maui project. Unfortunately it is very low quality video and I have no idea how to get a higher quality stream. I think it just grabs to lowest quality stream and returns it? I'm not sure. I have no idea how to just grab the highest quality. I have tried the other methods outlined in YoutubeExplode but all of them error out with stream being incompatible. The one I have above works but it's so low quality. In chrome the video quality has a bunch of stream options and I was hoping this addon would allow me to choose the quality some how. I imagine this is user error but I have been unable to figure it out. Any advice would be apprecited. It is an ongoing live stream by a podcast network.

**Comment by Tyrrrz** [comment, marked as answer]
Typically, the HLS manifest should have references to streams of various quality. You can test the URL out in VLC or some other video player. That means that whatever layer you're using in MAUI to play the stream, should be able to parse the manifest correctly and provide a way to switch qualities.

**Reply by ne0rrmatrix** [reply]
Ty That clarifies what I need to do. Appreciate. Atm I'm pretty sure media element has no support for that. I will have to create a function or class to parse the m3u8 file and extract the data. Not that hard to do really. I have tested it by doing it by hand and testing output. Works as expected. 

What surprised me is media element parsing the file at all and playing it back. It looks like it is just grabbing the first element and playing that back as a default way of dealing with the file. At one point it was explicitly stated this format was not currently supported at all. But now I have a path forward and can do everything myself so ty for the info and it was really useful.
